File: part1_Selenium/python/.ActionExamples/findElements.py
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from selenium import webdriver
driver = webdriver.Chrome(ChromeDriverManager().install())

# Id
driver.find_element_by_id("some id")
driver.find_element(By.ID, "some id")

# XPATH
driver.find_element_by_xpath("some xpath")
driver.find_element(By.XPATH, "some xpath")

# Link Text
driver.find_element_by_link_text("some link text")
driver.find_element(By.LINK_TEXT, "some link text")

# Partial Link Text
driver.find_element_by_partial_link_text("some partial link text")
driver.find_element(By.PARTIAL_LINK_TEXT, "some partial link text")

# CSS Selector
driver.find_element_by_css_selector("some css")
driver.find_element(By.CSS_SELECTOR, "some css")

# Class Name
driver.find_element_by_class_name("some class name")
driver.find_element(By.CLASS_NAME, "some class name")

# Name
driver.find_element_by_name("some name")
driver.find_element(By.NAME, "some name")

# Find all elements with matching selector
driver.find_elements_by_xpath("some xpath")
driver.find_elements(By.XPATH, "some xpath")

File: part1_Selenium/javascript/.ActionExamples/findElements.js
const {Builder} = require('selenium-webdriver');

async function main() {
    let driver = new Builder()
        .forBrowser('chrome')
        .usingServer('http://localhost:4444/wd/hub')
        .build();

    // Id
    driver.findElement(By.id("some id"));

    // XPATH
    driver.findElement(By.xpath("some xpath"));

    // Link Text
    driver.findElement(By.linkText("some link text"));

    // Partial Link Text
    driver.findElement(By.partialLinkText("some partial link text"));

    // CSS Selector
    driver.findElement(By.css("some css"));

    // Class Name
    driver.findElement(By.className("some class name"));

    // Name
    driver.findElement(By.name("some name"));

    // Find all elements with matching selector
    driver.findElements(By.xpath("some xpath"));
}